pkg/types/baremetal/validation/platform.go

```diff
@@ -404,23 +405,26 @@ func ValidatePlatform(p *baremetal.Platform, n *types.Networking, fldPath *field
 		}
 	}
 
-	if p.Hosts == nil {
+	agentBasedInstallation := len(os.Args) > 1 && os.Args[1] == "agent"
```

Comment:
On a second thought, this approach seems to me too specific and fragile. 
Is there anything that prevents us from skipping _entirely_ the baremetal validation (with a proper validation msg) - until we're going to really use it?

Reply:
We need to validate the VIPs somehow.
Also, if we're hoping that people can use the same install-config for both IPI and agent installs, then it doesn't make sense to allow them to put invalid data in it, even if it's unused.

An alternative to looking at `os.Args` might be to put some flag in the InstallConfig struct that gets set by `OptionalInstallConfig.Load()`. That isn't any less horrible, but it does have the advantage of working the same way in unit tests.